.github/workflows/ci.yml

```diff
@@ -5,54 +5,62 @@
 # the LICENSE file found in the root directory of this source tree.
 #
 
-name: Build
-
+name: CI
 on:
   push:
     branches:
       - 'main'
-
     tags:
       - '*'
-
   pull_request:
     branches:
       - '*'
 
 jobs:
-  build:
+  ghidra-script-gradle-tests:
+    strategy:
+      matrix:
+        image-version: [22.04]
+    runs-on: ubuntu-${{ matrix.image-version }}
+    timeout-minutes: 20
+    env:
+      CI: true
+    steps:
+      - name: Clean Docker to prevent space issues
+        run: |
+          docker system prune -af --volumes || true
+      - name: Clone the Patchestry repository
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 1
+      - name: Build the Ghidra testing environment and run tests in it via Gradle
+        run: |
+          DOCKER_BUILDKIT=1 docker build -t trailofbits/decompile-test:latest -f test/decompile-test.dockerfile .
+          docker run -it trailofbits/decompile-test
+
+  llvm-build-and-test:
     strategy:
       matrix:
         llvm-version: [20]
         image-version: [22.04]
         build-type: [Release, Debug]
         sanitizers: [OFF]
-
     runs-on: ubuntu-${{ matrix.image-version }}
-    timeout-minutes: 90
+    timeout-minutes: 20
     container:
       image:
         ghcr.io/lifting-bits/patchestry-ubuntu-${{ matrix.image-version }}-llvm-${{ matrix.llvm-version }}-dev:latest
-
```

Comment:
LIT test is running in a docker container that is inside docker. Removing `dind` service may cause issue with running test docker container.